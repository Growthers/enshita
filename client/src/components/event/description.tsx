import React from "react";

import styles from "~/styles/event/description.module.scss";

type Props = {
  description: string;
};

const EventDescription: React.FC<Props> = props => {
  const { description } = props;
  // descriptionの中からURLを抽出して配列に
  const urlPattern = /https?:\/\/[\w!?/+\-_~;.,*&@#$%=']+/gim;
  const URL = description.match(urlPattern) ?? [];

  return (
    <p className={styles["enshita-event-description-content"]}>
      {description
        // APIからは\\nが返ってくるので\nに置き換える
        .replace(/\\n/g, "\n")
        // URLで区切る（コンテンツの配列）
        .split(urlPattern)
        // contentとURLを交互に挿入
        // content0, url0, content1, url1, content2...
        .map((content, index) => (
          <span key={content}>
            {content}
            {URL[index] && (
              <a
                className={styles["enshita-event-description-link"]}
                href={URL[index]}
                target="_blank"
                rel="noopener noreferrer"
              >
                {URL[index]}
              </a>
            )}
          </span>
        ))}
    </p>
  );
};

export default EventDescription;
