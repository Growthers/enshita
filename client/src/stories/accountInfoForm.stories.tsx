// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
