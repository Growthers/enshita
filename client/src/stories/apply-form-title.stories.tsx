import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
