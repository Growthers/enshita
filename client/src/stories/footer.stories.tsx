// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
