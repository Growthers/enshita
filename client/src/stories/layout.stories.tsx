import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",

  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = () => (
  <Layout>
    <div>hogehoge</div>
  </Layout>
);
