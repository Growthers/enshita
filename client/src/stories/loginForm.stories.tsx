// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import LoginForm from "~/components/loginForm";

type T = typeof LoginForm;
type Story = ComponentStoryObj<T>;

export default {
  title: "LoginForm",
  component: LoginForm,
} as ComponentMeta<T>;

export const Default: Story = {};
