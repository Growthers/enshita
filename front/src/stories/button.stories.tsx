import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Button from "~/components/button";

type T = typeof Button;
type Story = ComponentStoryObj<T>;

export default {
  title: "Button",
  args: { children: <div>hogehoge</div> },
  component: Button,
} as ComponentMeta<T>;

export const Default: Story = {};
