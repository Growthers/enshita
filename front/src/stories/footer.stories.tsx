import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
