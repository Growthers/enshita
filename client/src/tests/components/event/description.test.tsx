import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/event/description.stories";

const { Default, SimpleText, URL } = composeStories(stories);

describe("(component) EventDescription", () => {
  test("Snap Shot - Simple Text", () => {
    const { container } = render(<SimpleText />);
    expect(container).toMatchSnapshot();
  });

  test("Snap Shot - Contain URL", () => {
    const { container } = render(<URL />);
    expect(container).toMatchSnapshot();
  });

  test("Snap Shot - Contain New Line and URL", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
