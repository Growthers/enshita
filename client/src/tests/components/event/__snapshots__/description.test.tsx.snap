// Vitest Snapshot v1

exports[`(component) EventDescription > Snap Shot - Contain New Line and URL 1`] = `
<div>
  <p
    class="_enshita-event-description-content_dpaga_1"
  >
    <span>
      イベントの説明

URL↓

      <a
        class="_enshita-event-description-link_dpaga_10"
        href="https://growthers.dev"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://growthers.dev
      </a>
    </span>
    <span />
  </p>
</div>
`;

exports[`(component) EventDescription > Snap Shot - Contain URL 1`] = `
<div>
  <p
    class="_enshita-event-description-content_dpaga_1"
  >
    <span>
      URL: 
      <a
        class="_enshita-event-description-link_dpaga_10"
        href="https://growthers.dev"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://growthers.dev
      </a>
    </span>
    <span />
  </p>
</div>
`;

exports[`(component) EventDescription > Snap Shot - Simple Text 1`] = `
<div>
  <p
    class="_enshita-event-description-content_dpaga_1"
  >
    <span>
      イベントの説明
    </span>
  </p>
</div>
`;
